CMakeLists.txt

```diff
@@ -74,6 +92,9 @@ catkin_package(
 # Thus, we add -fPIC here.
 # This allows us to still include this in any shared libraries we create.
 add_compile_options(-fPIC)
+# removed any pre-defined `WIN32_LEAN_AND_MEAN`.
+# Otherwise, many symbols will be reported as missing.
+remove_definitions(-DWIN32_LEAN_AND_MEAN)
```

Comment:
Would there be an alternative? AFAIR you could include the specific headers in the source files explicitely, but that would mean touching original SOEM source files. I'd like to avoid this as much as possible...

Reply:
Yes, I'd like to avoid to touch the SOEM code as much as possible too, so that's why I am doing this override from this root `CMakeLists.txt`.

Reply:
fine for me :+1: